vermouth/rcsu/get_go.py

```diff
@@ -0,0 +1,164 @@
+
+        for contact in self.contact_map:
+            resIDA, chainA, resIDB, chainB = contact
+            # identify the contact in the residue graph based on
+            # chain ID and resid
+            resA = chain_id_to_resnode[(chainA, resIDA)]
+            resB = chain_id_to_resnode[(chainB, resIDB)]
+            # make sure that both residues are not connected
+            # note: contacts should be symmteric so we should be
+            # able to get rid of the second if clause
+            if resB not in connected_pairs[resA] and resA not in connected_pairs[resB]:
+                # now we lookup the backbone nodes within the residue contact
+                bb_node_A = _get_bb_nodes(molecule, self.res_graph.nodes[resA]['graph'].nodes)
+                bb_node_B = _get_bb_nodes(molecule, self.res_graph.nodes[resB]['graph'].nodes)
+                # compute the distance between bb-beads
+                dist = np.linalg.norm(molecule.nodes[bb_node_A]['position'] -
+                                      molecule.nodes[bb_node_B]['position'])
+                # verify that the distance between BB-beads satisfies the
+                # cut-off criteria
+                if dist > self.cutoff_short or dist < self.cutoff_long:
+                    # compute the LJ sigma paramter for this contact
+                    sigma = dist / 1.12246204830
+                    # find the go virtual-sites for this residue
+                    # probably can be done smarter but mehhhh
+                    nodeA = _get_go_type(molecule, resid=resIDA, chain=chainA, prefix=self.prefix)
+                    nodeB = _get_go_type(molecule, resid=resIDB, chain=chainB, prefix=self.prefix)
+                    # write itp file with go interaction parameters
+                    self.itp_file.write(f"{nodeA} {nodeB} 1 {sigma:3.8F} {self.go_eps:3.8F} ; structural Go bias\n")
+                    # generate the exclusions between pairs of bb beads
+                    excl = Interaction(atoms=(bb_node_A, bb_node_B), parameters=[], meta={"group": "Go model exclusion"})
+                    molecule.interactions['exclusions'].append(excl)
+
+    def write_water_bias_auto(self, molecule):
+        """
+        Automatically calculate the water bias depending on the secondary structure.
+        """
+        self.itp_file.write('; additional Lennard-Jones interaction between virtual BB bead and W\n')
+        for res_node in self.res_graph.nodes:
+            resid = self.res_graph.nodes[res_node]['resid']
+            chain = self.res_graph.nodes[res_node]['chain']
+            vs_go_node = _get_go_type(molecule, resid=resid, chain=chain, prefix=self.prefix)
+            sec_struc = self.res_graph.nodes[res_node]['sec_struc']
+            eps = self.water_bias.get(sec_struc, 0.0)
+            if self.res_graph.nodes[res_node]['resname'] in ['GLY', 'ALA', 'VAL', 'PRO']:
+                sigma = 0.430
+            else:
+                sigma = 0.470
```

Comment:
This mapping needs to become a defaultdict somewhere. Either toplevel of this module/file, or the class, or the instance.
I'm leaning towards class at the moment.

Reply:
sigma values are now force-field variables, where the bead type determines the size. That is rather robust because Martini2 and Martini3 follow this principle with the tiny exemption of non-canonical beads. However, for the protein model, we do not care about those.